src/sysext/sysext.c

```diff
@@ -509,14 +509,15 @@ static int merge_subprocess(Hashmap *images, const char *workspace) {
                 }
 
                 /* Now that we can look into the extension image, let's see of the OS version is compatible */
-                r = parse_os_release(
+                r = parse_extension_release(
                                 p,
+                                img->name,
                                 "ID", &extension_os_release_id,
                                 "VERSION_ID", &extension_os_release_version_id,
                                 "SYSEXT_LEVEL", &extension_os_release_sysext_level,
                                 NULL);
                 if (r == -ENOENT)
-                        log_notice_errno(r, "Extension '%s' carries no os-release data, not checking for version compatibility.", img->name);
+                        return log_notice_errno(r, "Extension '%s' carries no extension-release data, ignoring extension.", img->name);
```

Comment:
if this is ignored you need to `continue` here, not `return`

Reply:
the extension is ignored if there's no extension-release - I thought we wanted to make it mandatory, to be at least present

Reply:
but if you use return here, then the whole discovery of extensions is terminated, and the whole thing fails. if you just want to ignore the one extension we look at, use `continue`

Reply:
agh, sorry, of course - will fix